src/test/rules/spaceInParensRuleTests.ts

```diff
@@ -0,0 +1,514 @@
+
+  { code: 'foo({ bar: "baz" }, [ 1, 2 ])', options: ['always', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo({\nbar: "baz"\n}, [\n1,\n2\n])', options: ['always', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo(); bar({bar:"baz"}); baz([1,2])', options: ['always', { exceptions: ['{}', '[]', '()'] }] },
+  { code: 'foo( { bar: "baz" }, [ 1, 2 ] )', options: ['never', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo( {\nbar: "baz"\n}, [\n1,\n2\n] )', options: ['never', { exceptions: ['{}', '[]'] }] },
+  { code: 'foo( ); bar( {bar:"baz"} ); baz( [1,2] )', options: ['never', { exceptions: ['{}', '[]', 'empty'] }] },
+
+  // faulty exceptions option
+  { code: 'foo( { bar: "baz" } )', options: ['always', { exceptions: [] }] }
+]);
+
+ruleTester.addTestGroup('invalid', 'should fail invalid', [
+  {
+    code: 'foo( bar)',
+    output: 'foo( bar )',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 0, column: 8 }])
+  },
+  {
+    code: 'foo(bar)',
+    output: 'foo( bar )',
+    options: ['always'],
+    errors: expecting([
+    { message: MISSING_SPACE_ERROR, line: 0, column: 4 },
+    { message: MISSING_SPACE_ERROR, line: 0, column: 7 }
+    ])
+  },
+  {
+    code: 'var x = ( 1 + 2) * 3',
+    output: 'var x = ( 1 + 2 ) * 3',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 0, column: 15 }])
+  },
+  {
+    code: 'var x = (1 + 2 ) * 3',
+    output: 'var x = ( 1 + 2 ) * 3',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 0, column: 9 }])
+  },
+  {
+    code: 'foo\n(bar\n)\n',
+    output: 'foo\n( bar\n)\n',
+    options: ['always'],
+    errors: expecting([{ message: MISSING_SPACE_ERROR, line: 1, column: 1 }])
+  },
+  {
+    code: 'bar(baz )',
+    output: 'bar(baz)',
+    options: ['never'],
+    errors: expecting([ { message: REJECTED_SPACE_ERROR, line: 0, column: 8 } ])
+  },
+  {
+    code: 'bar( baz )',
+    output: 'bar(baz)',
+    options: ['never'],
+    errors: expecting([
+    { message: REJECTED_SPACE_ERROR, line: 0, column: 4 },
+    { message: REJECTED_SPACE_ERROR, line: 0, column: 9 }
+    ])
+  },
+  {
+    code: 'var x = ( 4 + 5) * 6',
+    output: 'var x = (4 + 5) * 6',
+    options: ['never'],
+    errors: expecting([ { message: REJECTED_SPACE_ERROR, line: 0, column: 9 } ])
+  },
+  {
+    code: 'var x = (4 + 5 ) * 6',
+    output: 'var x = (4 + 5) * 6',
+    options: ['never'],
+    errors: expecting([ { message: REJECTED_SPACE_ERROR, line: 0, column: 15 } ])
+  },
+  {
+    code: 'new MyClass( hey)',
+    output: 'new MyClass( hey )',
+    options: ['always'],
+    errors: expecting([ { message: MISSING_SPACE_ERROR, line: 0, column: 16 } ])
+  },
+  {
+    code: 'new MyClass( hey)',
+    output: 'new MyClass(hey)',
+    options: ['never'],
+    errors: expecting([ { message: REJECTED_SPACE_ERROR, line: 0, column: 12 } ])
+  },
+
+  // classes
```

Comment:
Each of these comments mark a group. Lets separate them into their own

```
ruleTester.addTestGroup('test-id', 'description', [
```

In this case you can name it

```
ruleTester.addTestGroup('invalid-classes', 'should fail when declaring invalid spaces in methods', [
```

This will make it easier to find out which tests break in the future.